Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -533,11 +570,19 @@
         if (NSArray *matchPatterns = script[matchesKey])
             parameters.matchPatterns = makeVector<String>(matchPatterns);
 
-        if (NSArray *excludedMatchPatterns = script[excludeMatchesKey])
-            parameters.excludedMatchPatterns = makeVector<String>(excludedMatchPatterns);
+        if (NSArray *excludeMatchPatterns = script[excludeMatchesKey])
+            parameters.excludeMatchPatterns = makeVector<String>(excludeMatchPatterns);
+
+        if (isRegisteringScript || script[allFramesKey])
+            parameters.allFrames = boolForKey(script, allFramesKey, false);
 
-        if ([script[worldKey] isEqualToString:mainWorld])
-            parameters.world = WebExtensionContentWorldType::Main;
+        if (isRegisteringScript || script[persistAcrossSessionsKey])
+            parameters.persistent = boolForKey(script, persistAcrossSessionsKey, true);
+
+        if (NSString *world = script[worldKey])
+            parameters.forMainWorld = [world isEqualToString:mainWorld];
+        else if (isRegisteringScript)
+            parameters.forMainWorld = false;
```

Comment:
I think `WebExtensionContentWorldType` is still better than a `forMainWorld` `bool`.

Reply:
yea I was going back and forth between what to use here. `WebExtensionContentWorldType` is probably better in the case other worlds are added